x/ecocredit/msgs.go

```diff
@@ -390,31 +392,36 @@ func (m MsgSell) GetSignBytes() []byte {
 
 // ValidateBasic does a sanity check on the provided data.
 func (m *MsgSell) ValidateBasic() error {
+	now := time.Now()
```

Comment:
i think this could cause some non-determinism. if we create a sell order in the US to cancel tomorrow, some of the global validators might already technically be in _tomorrow_, and would error this, whereas US validators would pass it.

Reply:
The checks now take place within the server implementation methods and use block time.